Source/Core/Core/MemTools.cpp

```diff
@@ -264,10 +267,38 @@ static void sigsegv_handler(int sig, siginfo_t* info, void* raw_context)
                                  ))
   {
     // retry and crash
-    signal(SIGSEGV, SIG_DFL);
-#ifdef __APPLE__
-    signal(SIGBUS, SIG_DFL);
-#endif
+    // According to the sigaction man page, if sa_flags "SA_SIGINFO" is set to the sigaction
+    // function pointer, otherwise sa_handler contains one of:
+    // SIG_DEF: The 'default' action is performed
+    // SIG_IGN: The signal is ignored
+    // Any other value is a function pointer to a signal handler
+
+    struct sigaction* old_sa;
```

Comment:
can this be combined into one line, `struct sigaction* old_sa = sig == SIGSEGV ? &old_sa_segv : &old_sa_bus`?